dag_in_context/src/optimizations/passthrough.egg

```diff
@@ -31,12 +32,15 @@
       :ruleset always-run)
 
 ;; Pass through if arguments
-(rule ((= lhs (Get if i))
+(rule ((= if_get (Get if i))
        (= if (If pred inputs then_ else_))
-       (= (Get else_ i) (Get (Arg _) j))
-       (= (Get then_ i) (Get (Arg _) j))
-       (= passed-through (Get inputs j)))
-      ((union lhs passed-through))
+       (= (Get then_ i) (Get (InContext (InIf true pred inputs) (Arg ty)) j))
```

Comment:
Along the same lines as alex, I think you can match on any old context in here- doesn't need to be InIf